### PR TITLE
[master] Add debug log of the image format in guest_deploy

### PR DESCRIPTION
**Pull request by SeanHQF**
So that we can have more information to debug once there is problem
happened.

Signed-off-by: QingFeng Hao <haoqf@linux.vnet.ibm.com>

### CODE REVIEW

File: data/sudoers-zvmsdk [hunk 1]
```diff
@@ -1 +1 @@
-zvmsdk ALL = (ALL) NOPASSWD:/sbin/vmcp, /opt/zthin/bin/smcli, /sbin/chccwdev, /sbin/cio_ignore, /sbin/fdasd, /sbin/fdisk, /usr/sbin/vmur, /bin/mount, /bin/umount, /sbin/mkfs, /sbin/mkfs.xfs, /usr/sbin/mkswap, /sbin/dasdfmt, /opt/zthin/bin/unpackdiskimage, /opt/zthin/bin/creatediskimage, /opt/zthin/bin/linkdiskandbringonline, /opt/zthin/bin/offlinediskanddetach, /opt/zthin/bin/IUCV/iucvclnt, /opt/zthin/bin/refresh_bootmap
+zvmsdk ALL = (ALL) NOPASSWD:/sbin/vmcp, /opt/zthin/bin/smcli, /sbin/chccwdev, /sbin/cio_ignore, /sbin/fdasd, /sbin/fdisk, /usr/sbin/vmur, /bin/mount, /bin/umount, /sbin/mkfs, /sbin/mkfs.xfs, /usr/sbin/mkswap, /sbin/dasdfmt, /opt/zthin/bin/unpackdiskimage, /opt/zthin/bin/creatediskimage, /opt/zthin/bin/linkdiskandbringonline, /opt/zthin/bin/offlinediskanddetach, /opt/zthin/bin/IUCV/iucvclnt, /opt/zthin/bin/refresh_bootmap, /usr/bin/hexdump
```
Comment (jichenjc) [inline data/sudoers-zvmsdk:1]: this hexdump command need sudo privilege?

File: zvmsdk/smtclient.py [hunk 2]
```diff
@@ -863,6 +863,17 @@ def guest_deploy(self, userid, image_name, transportfiles=None,
...
+        image_file = '/'.join([self._get_image_path_by_name(image_name),
+                               CONF.zvm.user_root_vdev])
+        # Run 'hexdump -C -n 64 redhat84_eckd_iso_new_zvmguestconfigure.img' to
+        # parse the image header
+        cmd = ['sudo', '/usr/bin/hexdump', '-C', '-n', '64', image_file]
+        with zvmutils.expect_and_reraise_internal_error(modID='guest'):
+            (rc, output) = zvmutils.execute(cmd)
+            msg = ('Image header info in guest_deploy: rc: %d, header: %s'
+                   % (rc, output))
+            LOG.info(msg)
```
Comment (jichenjc) [inline zvmsdk/smtclient.py:875]: can you paste the output through your test for review as reference?

File: zvmsdk/smtclient.py [hunk 3]
```diff
@@ -863,6 +863,17 @@ def guest_deploy(self, userid, image_name, transportfiles=None,
                      remotehost=None, vdev=None, skipdiskcopy=False):
         """ Deploy image and punch config driver to target """
         # (TODO: add the support of multiple disks deploy)
+        image_file = '/'.join([self._get_image_path_by_name(image_name),
+                               CONF.zvm.user_root_vdev])
+        # Run 'hexdump -C -n 64 redhat84_eckd_iso_new_zvmguestconfigure.img' to
```
Comment (jichenjc) [inline zvmsdk/smtclient.py:868]: I Think redhat84_eckd_iso_new_zvmguestconfigure.img is just an example?

File: zvmsdk/smtclient.py (hunk 2 above)
Comment (SeanHQF) [reply zvmsdk/smtclient.py:875]: Pls check https://github.com/openmainframeproject/feilong/pull/511#issuecomment-911078946

File: zvmsdk/smtclient.py (hunk 3 above)
Comment (SeanHQF) [reply zvmsdk/smtclient.py:868]: Yes, an example.

File: zvmsdk/smtclient.py (hunk 2 above)
Comment (SeanHQF) [reply zvmsdk/smtclient.py:875]: latest output:
```
[2021-09-02 01:53:19] [INFO] Image header info in guest_deploy: rc: 0, header:
00000000  78 43 41 54 20 43 4b 44  20 44 69 73 6b 20 49 6d  |xCAT CKD Disk Im|
00000010  61 67 65 3a 20 20 20 20  20 20 20 31 34 35 36 34  |age:       14564|
00000020  20 43 59 4c 20 48 4c 65  6e 3a 20 30 30 35 35 20  | CYL HLen: 0055 |
00000030  47 5a 49 50 3a 20 36 20  20 20 20 20 20 20 20 20  |GZIP: 6         |
00000040
```

File: data/sudoers-zvmsdk (hunk 1 above)
Comment (SeanHQF) [reply data/sudoers-zvmsdk]: no, it doesn't need, thanks and updated

File: zvmsdk/smtclient.py (hunk 2 above)
Comment (dongyanyang) [reply zvmsdk/smtclient.py:875]: @SeanHQF @jichenjc 
These logic should be added into the else of if skipdiskcopy... 
for RHCOS and BFV, there will be no image unpack action in the guest_deploy.
For BFV, we've opened an issue for the regression error, you can check.
@bjhuangr please correct if I'm wrong about RHCOS.

Comment (bjhuangr) [reply zvmsdk/smtclient.py:875]: yes, log hexdump info here does not help for RHCOS deployment. but also no harm ..

Comment (dongyanyang) [reply zvmsdk/smtclient.py:875]: with this change, BFV always fail as the image file path not exist.

Comment (jichenjc) [reply zvmsdk/smtclient.py:875]: the follow up PR merge (#517 )

**Pull request merged**